packages/nuxt/src/pages/runtime/page.ts

```diff
@@ -81,29 +94,35 @@ export default defineComponent({
           const key = generateRouteKey(routeProps, props.pageKey)
 
           const hasTransition = !!(props.transition ?? routeProps.route.meta.pageTransition ?? defaultPageTransition)
+          trackRootNodes = hasTransition
           const transitionProps = hasTransition && _mergeTransitionProps([
             props.transition,
             routeProps.route.meta.pageTransition,
             defaultPageTransition,
             { onAfterLeave: () => { nuxtApp.callHook('page:transition:finish', routeProps.Component) } }
           ].filter(Boolean))
 
-          vnode = _wrapIf(Transition, hasTransition && transitionProps,
-            wrapInKeepAlive(props.keepalive ?? routeProps.route.meta.keepalive ?? (defaultKeepaliveConfig as KeepAliveProps), h(Suspense, {
-              suspensible: true,
-              onPending: () => nuxtApp.callHook('page:start', routeProps.Component),
-              onResolve: () => { nextTick(() => nuxtApp.callHook('page:finish', routeProps.Component).finally(done)) }
-            }, {
-              default: () => h(RouteProvider, {
+          componentVNode = h(routeProps.Component, { ref: pageRef, key: key || undefined })
+          vnode = h(RouteProvider, {
+            route: routeProps.route,
+            renderKey: key || undefined
+          }, _wrapIf(
+            Transition,
+            hasTransition && transitionProps,
+            wrapInKeepAlive(
+              props.keepalive ?? routeProps.route.meta.keepalive ?? defaultKeepaliveConfig as KeepAliveProps,
+              h(Suspense, {
+                suspensible: true,
                 key: key || undefined,
-                vnode: routeProps.Component,
-                route: routeProps.route,
-                renderKey: key || undefined,
-                trackRootNodes: hasTransition,
-                vnodeRef: pageRef
-              })
-            })
-            )).default()
+                onPending: () => nuxtApp.callHook('page:start', routeProps.Component),
+                onResolve: () => {
+                  nextTick(() => nuxtApp.callHook('page:finish', routeProps.Component).finally(done))
+                }
+              }, componentVNode)
+            )
+          ))
+
+          ;(vnode.type as any).__hmrId = (componentVNode.type as any).__hmrId
```

Comment:
we should probably only do this in dev mode:
```suggestion
          if (import.meta.dev) {
            (vnode.type as any).__hmrId = (componentVNode.type as any).__hmrId
          }
```